### DISCUSSION TITLE
let clangd know the correct working directory

### DISCUSSION THREAD

**Discussion by Freed-Wu** [clangd/clangd#1570] [Q&A] [answered]
This is an example:

`common/a.h`

```c
int f(int);
```

`common/a.c`
```c
#include "common/a.h"

int g() {
	return f(1);
}
```

`vi common/a.c`, it will get

```
clang pp_file_not_found: 'common/a.h' file not found
```

The screenshot is for coc-clangd:

![screen-2023-04-03-17-18-57](https://user-images.githubusercontent.com/32936898/229468811-06117a48-a9c0-4b8d-a8e9-945c5645b302.jpg)

If I create `.clangd`:

```yaml
---
CompileFlags:
  Add:
    - -I.
```

It will not work. If I change it to

```yaml
---
CompileFlags:
  Add:
    - -I..
```

the error about pp_file_not_found will disappear.

So, how can I let clangd know the working directory?
I hope `clangd` know the directory contains `.git` is true `.`, not the directory containing file (which is `common` containing `common/a.c`).

TIA!

**Comment by i-ky** [comment, marked as answer]
> So, how can I let clangd know the working directory?

In `compile_commands.json` there can be `directory` for each compiler invocation, see [specification](https://clang.llvm.org/docs/JSONCompilationDatabase.html).
